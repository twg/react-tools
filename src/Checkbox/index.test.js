/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox } from './'

const noop = () => {}
describe('Checkbox', () => {
  it('renders an unchecked checkbox', () => {
    let component = renderer.create(
      <Checkbox
        value={false}
        onChange={noop}
        id="unchecked"
        label="Unchecked Example"
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders an checked checkbox', () => {
    let component = renderer.create(
      <Checkbox
        value={true}
        onChange={noop}
        id="checked"
        label="Checked Example"
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
