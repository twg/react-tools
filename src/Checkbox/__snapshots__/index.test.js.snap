// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders an checked checkbox 1`] = `
<div
  className="checkBoxContainer,"
>
  <label
    className="labelCheckbox"
    htmlFor="checked"
  >
    <input
      checked={true}
      className="invisible"
      disabled={false}
      name="checked"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="checkbox"
    >
      <svg
        height="20px"
        viewBox="0 0 20 20"
        width="20px"
      >
        <path
          d="M3,1 L17,1 L17,1 C18.1045695,1 19,1.8954305 19,3 L19,17 L19,17 C19,18.1045695 18.1045695,19 17,19 L3,19 L3,19 C1.8954305,19 1,18.1045695 1,17 L1,3 L1,3 C1,1.8954305 1.8954305,1 3,1 Z"
        />
        <polyline
          points="4 11 8 15 16 6"
        />
      </svg>
    </div>
    <span>
      Checked Example
    </span>
  </label>
</div>
`;

exports[`Checkbox renders an unchecked checkbox 1`] = `
<div
  className="checkBoxContainer,"
>
  <label
    className="labelCheckbox"
    htmlFor="unchecked"
  >
    <input
      checked={false}
      className="invisible"
      disabled={false}
      name="unchecked"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="checkbox"
    >
      <svg
        height="20px"
        viewBox="0 0 20 20"
        width="20px"
      >
        <path
          d="M3,1 L17,1 L17,1 C18.1045695,1 19,1.8954305 19,3 L19,17 L19,17 C19,18.1045695 18.1045695,19 17,19 L3,19 L3,19 C1.8954305,19 1,18.1045695 1,17 L1,3 L1,3 C1,1.8954305 1.8954305,1 3,1 Z"
        />
        <polyline
          points="4 11 8 15 16 6"
        />
      </svg>
    </div>
    <span>
      Unchecked Example
    </span>
  </label>
</div>
`;
